### DISCUSSION TITLE
Welcome to Z80-Retro Discussions!

### DISCUSSION THREAD

**Discussion by peterw8102** [peterw8102/Z80-Retro#8] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by JaapvanGanswijk** [comment]
Hi Peter, I ordered your Z80 board from Ebay and got it today. Still studying it, but I saw that you wanted to hear back from buyers of the board. I have quite a lot of Z80-experience starting with a Tandy Radio Shack TRS80 home computer in 1979. I wrote disassemblers, assemblers, linkers and even a compiler for it (of I remember correctly ;-). Later!

**Comment by JaapvanGanswijk** [comment]
Oops later I saw that we had already communicated on the EasyEDA platform, but back then I had only seen your design and hadn't bought an actual PCB yet, but now I have, so I studied your design some more and are very impressed with the 74LS670's forming the MMU, but we did a similar thing in our 6809-computer! Haven't checked if you have a separate supervisor state, but that might be done by the 7474 flipflop, if so the system might be able to run a kind of Unix! Exciting!

**Comment by peterw8102** [comment]
Hi Jaapvan, hope you have fun with the board. Let me know how you get on and if you have any problems. I haven't had a lot of time to work on it recently due to other commitments but hope that's going to change soon. I have some ideas I want to work on.

There's not really a usable supervisor mode. On boot the 74LS670s are effectively disabled so the first block of flash is mapped to address 0. When my code starts up it programmes the MMU registers, copies the flash code to RAM, confugures the MMU then pages in the RAM to page 0. I generally don't run from flash. You can switch off the MMU but then you're simply back to Flash mapped to the address space. There might be a more imaginative way to use this hardware so let me know if you can find one.

I haven't done this, but the [guy that works on Fuzix](https://github.com/EtchedPixels/FUZIX) has got that working on the emulated version on the board. Running that up on actual hardware is also on my list. He did that by [getting RomWBW running on the system](https://github.com/wwarthen/RomWBW) so that's also an option, but again not something I've looked at. RomWBW seems too complicated and monolithic for my interests.

All the best
Pete

**Reply by JaapvanGanswijk** [reply]
Thanks for your response! Will have to look into this some more. I think I'm in contact with the guy from RomWBW. Also wanted to check out that Fuzix. BTW. also found another Z80 board with 74670's, but haven't studied (or ordered) them yet. Don't remember how we did the supervisor mode on our 6809 system. The problem was that the 6809 has some test instructions that mess up the CPU even in user mode. :-( So if the processor started to execute data it would mess up the system. Was no problem when the users were running tested code, but when I developed programs I'd have to do a sync() at the start to deminish the damage. Sync() writes the cache to the harddisk.

**Comment by JaapvanGanswijk** [comment]
I still have to get all the components (and solder them), but you use a lot of different 7400 technologies like LS, F, HCT, AHCT, ACT and none. Is that necessary or can I use HC? I seem to remember that ACT and F are fast and I guess that everything having to do with the MMU has to be very fast, but all the other chips? HC doesn't use real TTL input levels, but might work anyway. Also still considering which chips to put in sockets...

**Reply by peterw8102** [reply]
I just looked at the BOM I have up and I can't see any TTL with no family. Some of the families are easier to get hold of than others, hence the mix. The ones listed are the ones I either already had or could get hold of. I bought everything from Mouser. A few suggestions. Don't use HC family chips - they are much slower and the levels aren't TTL compatible. I use a 14MHz clock so didn't want to take any chances. Faster is better.

The 74670 registers I could only get in HC or LS - go for LS. I did try the HC and was quite surprised they worked but I might have been lucky. On my current board I'm using 74LS670. ACT/AHCT would be my preference, then F (draw more power but fast) then HCT and finally LS if you can't get anything else.

**Reply by JaapvanGanswijk** [reply]
Thanks, that helps a lot! I'll first use HC for the slow parts (if I already have them) and won't use sockets and then I'll order faster IC's for the more critical parts.
